### ISSUE TITLE
The version check is limited to valid `appsody` commands

### ISSUE THREAD

**Issue by skoh7645** [appsody/appsody#412]
**Describe the bug**
The version check message is only shown for valid `appsody` commands, rather than on every invocation of `appsody`. What it be possible to change where the function is called to support this?

**To Reproduce**
1. Have an older version of appsody installed
2. Run 'appsody something" (any invalid `appsody` command)
3. No "please update CLI" message is shown

**Expected behavior**
Version check is completed on every invocation of `appsody`

**Actual behaviour**
Version check is only completed on an invocation of a valid `appsody` command

**If applicable please specify:**
 - CLI version: 0.4.6
 - Stack you are using:

**Comment by kylegc** [comment]
I agree this would be nice, but the implementation has a tricky catch22: lastVersionCheck is read/written from the users's config file which can be changed with the `--config` flag. But the `--config` flag won't be parsed if there are invalid command args.

Feel free to give this a shot, but in my opinion, this is probably not important enough to outweigh the hassle to resolve the situation above.
